catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/GeoPdfParser.java

```diff
@@ -0,0 +1,292 @@
+                        if (GEOGRAPHIC.equals(projectionType)) {
+                            COSArray neatlineArray = (COSArray) pdPage.getCOSObject()
+                                    .getObjectFromPath(LGIDICT + "/[" + i + "]/" + NEATLINE);
+                            String wktString = getWktFromNeatLine(lgidict,
+                                    neatlineArray,
+                                    toDoubleVisitor);
+                            polygons.add(wktString);
+                        } else {
+                            LOGGER.debug(
+                                    "Unsupported projection type {}.  Map Frame will be skipped.",
+                                    projectionType);
+                        }
+                    } else {
+                        LOGGER.debug(
+                                "No projection array found on the map frame.  Map Frame will be skipped.");
+                    }
+                }
+            // Handle One Map Frame
+            } else if (lgiDictObject instanceof COSDictionary) {
+                COSDictionary lgidict = (COSDictionary) lgiDictObject;
+                COSDictionary projectionArray = (COSDictionary) lgidict.getDictionaryObject(
+                        PROJECTION);
+                if (projectionArray != null) {
+                    String projectionType =
+                            ((COSString) projectionArray.getItem(PROJECTION_TYPE)).getString();
+                    if (GEOGRAPHIC.equals(projectionType)) {
+                        COSArray neatlineArray = (COSArray) pdPage.getCOSObject()
+                                .getObjectFromPath(LGIDICT + "/" + NEATLINE);
+                        if (neatlineArray == null) {
+                            neatlineArray = generateNeatLineFromPDFDimensions(pdPage);
+
+                        }
+                        polygons.add(getWktFromNeatLine(lgidict, neatlineArray, toDoubleVisitor));
+
+                    } else {
+                        LOGGER.debug("Unsupported projection type {}.  Map Frame will be skipped.",
+                                projectionType);
+                    }
+                } else {
+                    LOGGER.debug(
+                            "No projection array found on the map frame.  Map Frame will be skipped.");
+                }
+            }
+        }
+
+        if (polygons.size() == 0) {
+            LOGGER.debug(
+                    "No GeoPDF information found on PDF during transformation.  Metacard location will not be set.");
+            return null;
+        }
+
+        StringBuilder wkt = new StringBuilder();
+        if (polygons.size() == 1) {
+            wkt.append(POLYGON);
+            wkt.append(polygons.get(0));
+            wkt.append("))");
+        } else {
+            wkt.append(MULTIPOLYGON);
+            for (int i = 0; i < polygons.size(); i++) {
```

Comment:
Calling `get()` and `set()` repeatedly on a `LinkedList` will be slow if there are many polygons.

An alternate solution using streams:
```
return polygons.stream()
    .map(polygon -> "((" + polygon + "))")  // Could make a separate Function<String, String> to make this cleaner
    .collect(Collectors.joining(",", MULTIPOLYGON, ")"));
```